packages/walletconnect/src/index.ts

```diff
@@ -81,24 +94,26 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
-  private async isomorphicInitialize(
+  private isomorphicInitialize(
     desiredChainId: number | undefined = this.defaultChainId
-  ): Promise<MockWalletConnectProvider> {
+  ): Promise<WalletConnectProvider> {
     if (this.eagerConnection) return this.eagerConnection
 
-    const chains = desiredChainId ? orderToSetDefaultChain(this.chains, desiredChainId) : this.chains
-    const rpcMap = this.rpcMap ? await getRpcBestUrlMap(this.rpcMap, this.timeout) : undefined
-
     return (this.eagerConnection = import('@walletconnect/ethereum-provider').then(async (ethProviderModule) => {
+      const chains = desiredChainId ? orderToSetDefaultChain(this.chains, desiredChainId) : this.chains
+      const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
```

Comment:
I know this is contradictory to my last review, but I think this should be moved outside of the promise so it is not serial, otherwise you'll have to wait for the module load before beginning to request urls:

    const rpcMap = this.rpcMap ? getBestUrlMap(this.rpcMap, this.timeout) : undefined
    ...
      await rpcMap